streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -23,11 +23,13 @@
 import java.util.Map;
 
 /**
- * The window specifications used for joins.
+ * The window specifications used for joins. TODO extend
  */
 public class JoinWindows extends Windows<TumblingWindow> {
 
+    /** Maximum time difference for tuples that are before the join tuple. */
     public final long before;
```

Comment:
@guozhangwang: See previous question -- rename these fields to `beforeMs` and `afterMs`?